migrations/20190321232837_create_users_table.js

```diff
@@ -0,0 +1,29 @@
+
+exports.up = function(knex, Promise) {
+  return knex.schema.withSchema('public').createTable('users', function(table) {
+    table.increments('user_id');
+    table.string('username', 20).notNullable();
+    table.string('password', 100).notNullable();
+    table.string('email', 50).notNullable();
+    table.timestamp('date_created').defaultTo(knex.fn.now());
+
+    table.unique('username');
+    table.unique('email');
+  })
+}
+
+exports.down = function(knex, Promise) {
+  return knex.schema.dropTable('users');
+}
+
+/*
+CREATE TABLE IF NOT EXISTS users (
+	user_id			INT NOT NULL PRIMARY KEY AUTO_INCREMENT,
+    username		VARCHAR(20) NOT NULL,
+    `password`  	VARCHAR(100) NOT NULL,
+    email			VARCHAR(50) NOT NULL,
+    date_created	TIMESTAMP NOT NULL DEFAULT NOW(),
+    CONSTRAINT unique_user UNIQUE(username),
+    CONSTRAINT unique_umail UNIQUE(email)
+);
+*/
```

Comment:
End files with a newline. This is easy to miss. If you add your new `migrations/` folder to the args of the `lint` script in `package.json` then you can run `yarn lint` to run a code style and correctness check on all the js code. Running `yarn lint --fix` can automatically fix many simple bugs and style issues.

Reply:
Thanks for the info! Updated.